src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.c

```diff
@@ -1664,6 +1732,21 @@ static void lb_call_on_retry_timer_locked(grpc_exec_ctx *exec_ctx, void *arg,
   GRPC_LB_POLICY_WEAK_UNREF(exec_ctx, &glb_policy->base, "grpclb_retry_timer");
 }
 
+static void lb_on_fallback_timer_locked(grpc_exec_ctx *exec_ctx, void *arg,
+                                        grpc_error *error) {
+  glb_lb_policy *glb_policy = arg;
+  glb_policy->fallback_timer_active = false;
+  if (!glb_policy->shutting_down && error == GRPC_ERROR_NONE) {
+    if (GRPC_TRACER_ON(grpc_lb_glb_trace)) {
+      gpr_log(GPR_INFO, "Falling back to use backends from resolver (grpclb %p)",
+              (void *)glb_policy);
+    }
+    GPR_ASSERT(glb_policy->fallback_backend_addresses != NULL);
```

Comment:
I think there's a possible race condition here.  In `lb_on_response_received_locked()`, when we get the first serverlist, we cancel the timer and set `glb_policy->fallback_backend_addresses` to NULL.  However, if `lb_on_response_received_locked()` runs after the timer schedules this callback but before this callback runs, then `glb_policy->fallback_backend_addresses` will be NULL here.

I suggest removing this assertion and instead adding the following on line 1738:

```
// If we've already gotten a serverlist from the balancer, there's nothing to do here.
if (glb_policy->fallback_backend_addresses == NULL) return;
```